src/scripts/color-copy/color-copy.lua

```diff
@@ -0,0 +1,55 @@
+-- Abort if the API isn't available. Mouse events become available on Mudlet version 4.13.
+if addMouseEvent == nil then return end
+
+function rgbToHex(r,g,b)
+    local rgb = (r * 0x10000) + (g * 0x100) + b
+    return string.format("%x", rgb)
+end
+
+local baseAnsiMap = {"&x","&r","&g","&O","&b","&p","&c","&w","&z","&R","&G","&Y","&B","&P","&C","&W"}
+local colorCodes = {}
+
+-- Map basic ANSI set
+for i=0,15 do
+  local hex = rgbToHex(unpack(color_table["ansi_" .. string.format("%03d", i)]))
+  colorCodes[hex] = baseAnsiMap[i + 1]
+end
+
+-- Map the rest of 256 colors
+for i=16,255 do
+  local numStr = string.format("%03d", i)
+  local hex = rgbToHex(unpack(color_table["ansi_" .. numStr]))
+  colorCodes[hex] = "&" .. numStr
+end
+
+addMouseEvent("Copy with colors", "onCopyWithColors")
```

Comment:
If people upgrade the package or edit this, we'll end up registering this twice, right? We'd just want to make sure it doesn't happen again when the UI is reloaded, so you could just use a sentinel global variable for this and the event registration below.

Reply:
That API call will fail when trying to add an event with an identical name. So it should be good!

Reply:
Cool!

Reply:
Just to expand, this just ensures there's a link between mouse menu "Copy with colors" to "onCopyWithColors" event. If the event handler is updated the mouse event will use that. 

I've yet to write up that API in Mudlet wiki, but that's basically how it works. Kinda like creating databases, this just ensures the event is linked.